jboss/container/maven/default/bash/artifacts/opt/jboss/container/maven/default/maven.sh

```diff
@@ -155,11 +154,38 @@ function _add_maven_proxy() {
   fi
 }
 
+# Finds the environment variable  and returns its value if found.
+# Otherwise returns the default value if provided.
+#
+# Arguments:
+# $1 env variable name to check
+# $2 default value if environment variable was not set
 function _maven_find_env() {
   local var=${!1}
   echo "${var:-$2}"
 }
 
+# Finds the environment variable with the given prefix. If not found
+# the default value will be returned. If no prefix is provided will
+# rely on _maven_find_env
+#
+# Arguments
+#  - $1 prefix. Transformed to uppercase and replace - by _
+#  - $2 variable name. Prepended by "prefix_"
+#  - $3 default value if the variable is not defined
+function _maven_find_prefixed_env() {
+  local prefix=$1
+
+  if [[ -z $prefix ]]; then
+    _maven_find_env $2 $3
+  else
+    prefix=${prefix^^} # uppercase
+    prefix=${prefix//-/_} #replace - by _
+
+    local var_name=$prefix"_"$2
+    echo ${!var_name:-$3}
+  fi
+}
 # insert settings for mirrors/repository managers into settings.xml if supplied
 # internal function, use process_maven_settings_xml which applies all configuration
 function add_maven_mirrors() {
```

Comment:
I suggest:
instead of 205-207:
     `local mirror_id=$(_maven_find_env "${maven_mirror_prefix}_MAVEN_MIRROR_ID" "mirror${counter}")`
     `local mirror_url=$(_maven_find_env "${maven_mirror_prefix}_MAVEN_MIRROR_URL")`
     `local mirror_of=$(_maven_find_env "${maven_mirror_prefix}_MAVEN_MIRROR_OF" "external:*")`
 to use _maven_find_prefixed_env:
     `local mirror_id=$(_maven_find_prefixed_env "${maven_mirror_prefix}" "MAVEN_MIRROR_ID" "mirror${counter}")`
     `local mirror_url=$(_maven_find_prefixed_env "${maven_mirror_prefix}" "MAVEN_MIRROR_URL")`
     `local mirror_of=$(_maven_find_prefixed_env "${maven_mirror_prefix}" "MAVEN_MIRROR_OF" "external:*")`
because it can not read raw prefix here e.g. dev-one_MAVEN_MIRROR_URL even if this env exist. But it can read DEV_ONE_MAVEN_MIRROR_URL

Reply:
Good idea, but this was part of the original maven.sh code. It was not part of what I introduced. So whereas I think this could be an improvement, it shouldn't hold back this PR from getting processed.

Reply:
If user tries to use MAVEN_REPOS "dev-one,qe-two" than it is not possible to load MAVEN_MIRROR_URL .. I just copied all "examples" and did not get too far :) .. I agree and will proceed without this, it is not show-stopper and I will create ticket for this.